File: materiais/Algoritmos de Busca e Ordenação/Códigos/bubble_sort.py
def bubble_sort(L):
	swap = False
	while not swap:
		swap = True 
		for j in range(1,len(L)):
			if L[j-1] > L[j]:
				swap = False
				temp = L[j]
				L[j] = L[j-1]
				L[j-1] = temp
	return L 

L = [5,4,3,2,1]
print(bubble_sort(L))

File: Exemplos C/Sorting/bubble_sort.c
#include <stdio.h>
#include "swap.h"

// Programa que implementa o algoritmo Bubble Sort
// Adaptive & Stable
// Time -> Min: O(n) Max: O(n²)

// função que define o algoritmo bubble sort
// Recebe um array e o número de elementos desse array
void bubble_sort(int A[], int n)
{
    int i, j, flag=0;

    for(i=0;i<n-1;i++)
    {
        flag = 0;
        for(j=0;j<n-i-1;j++)
        {
            if(A[j] > A[j+1])
            {
                swap(&A[j],&A[j+1]);
                flag = 1;
            }
        }
        if(flag == 0)
            break;
    }
}

int main()
{
    int array[] = {3,10,13,27,2,1,8,15,99,180,4,5};
    int n = 12;
    bubble_sort(array,n);
    for(int i=0;i<n;i++)
        printf("%d\n",array[i]);
}